src/app-logic/web-channel.js

```diff
@@ -163,6 +259,25 @@ function _sendMessageWithResponse<Returns: MessageFromBrowser>(
 
     window.addEventListener('WebChannelMessageToContent', listener, true);
 
+    // Add the requestId to the message. The (...: any) is due to
+    // https://github.com/facebook/flow/issues/1488 which is probably fixed
+    // in flow v0.111.
+    const messageToBrowser: MessageToBrowser<B> = ({
+      requestId,
+      ...messageToBrowserBody,
+    }: any);
+
     _sendMessage(messageToBrowser);
   });
 }
+
+function _fixupOldResponseMessageIfNeeded(message: MixedObject) {
+  if (message.type === 'STATUS_RESPONSE') {
+    const { menuButtonIsEnabled } = message;
+    message.type = 'SUCCESS_RESPONSE';
+    message.response = { menuButtonIsEnabled };
+  } else if (message.type === 'ENABLE_MENU_BUTTON_DONE') {
+    message.type = 'SUCESS_RESPONSE';
```

Comment:
typo: `SUCCESS_RESPONSE`

This was still adding the button, but with an error in the console, and no change in the frontend UI.
(hence my request for a test covering this code :-) ).

Reply:
Nice catch! :)

Reply:
I've added a test for this. Unfortunately I had to make a choice between `triggerResponse` accepting `MixedObject`, and `triggerResponse` accepting `MessageFromBrowser<R>` and lying to the type system for legacy responses, because legacy responses are not covered by the `MessageFromBrowser` type. I chose the latter. Let me know if you'd prefer a different solution.